spot_wrapper/wrapper.py

```diff
@@ -2047,25 +2056,56 @@ def _upload_graph_and_snapshots(self, upload_filepath):
             )
         for waypoint in self._current_graph.waypoints:
             # Load the waypoint snapshots from disk.
-            with open(
-                upload_filepath + "/waypoint_snapshots/{}".format(waypoint.snapshot_id),
-                "rb",
-            ) as snapshot_file:
+            if len(waypoint.snapshot_id) == 0:
+                continue
+            waypoint_filepath = os.path.join(
+                upload_filepath, "waypoint_snapshots", waypoint.snapshot_id
+            )
+            if not os.path.exists(waypoint_filepath):
+                continue
+            with open(waypoint_filepath, "rb") as snapshot_file:
                 waypoint_snapshot = map_pb2.WaypointSnapshot()
                 waypoint_snapshot.ParseFromString(snapshot_file.read())
                 self._current_waypoint_snapshots[
                     waypoint_snapshot.id
                 ] = waypoint_snapshot
+
+                for fiducial in waypoint_snapshot.objects:
+                    if not fiducial.HasField("apriltag_properties"):
+                        continue
+
+                    str_id = str(fiducial.apriltag_properties.tag_id)
+                    if (
+                        str_id in self._current_anchored_world_objects
+                        and len(self._current_anchored_world_objects[str_id]) == 1
+                    ):
```

Comment:
and I think you should use the same style here too. Negate the conditional, replace the body with `continue`, and dedent the remainder.

Reply:
I think in this case because the block is at the end of the for loop, it is still readable? Also this for loop is pasted from BD's example code. Perhaps leaving it this way makes that obvious